File: 002.clj
(defn fib-seq
  [max]
  (loop [fibs [1 0]]
    (if (> (first fibs) max)
      (rest fibs)
      (recur (cons (+ (first fibs) (second fibs)) fibs)))))

(defn sum-even-fib
  [max]
  (reduce + (filter (fn [x] (= 1 (rem x 2)))
                    (fib-seq max))))

(println (sum-even-fib 4000000))

File: 002.erl
#!/usr/bin/env escript
% vim:set ts=4 sw=4:

main(_) ->
	Even = fun(X) -> ( X rem 2 =:= 0 ) end,
	EvenFib = lists:filter(Even, fibonacci(4000000)),
	io:format("~w~n", [lists:sum(EvenFib)]).

fibonacci(Max) -> fibonacci(Max, [1, 0]).
fibonacci(Max, [H|T]) when H > Max -> T;
fibonacci(Max, List) ->
	[H1|T] = List, [H2|_] = T,
	fibonacci(Max, [H1+H2|List]).
